src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,269 @@
+            None 
+        ''' 
+        self.destroy()
+
+    def __key_press_event_cb(self, window, event):
+        '''
+        key-press-event signal handler
+        Destroys the pop-up window on pressing Escape.
+
+        Args:
+           window (Gtk.Widget): the window which received the signal
+           event (Gdk.EventKey): the event which triggered the signal
+
+        Returns:
+            None
+        '''
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Escape':
+            self.destroy()
+
+    def __realize_cb(self, widget):
+        '''
+        realize signal handler
+        Sets window as a transient window for the window specified by parent.
+
+        Args:
+            widget (Gtk.Widget): the widget(window) which received the signal
+
+        Returns:
+            None
+        '''
+        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
+        window = self.get_window()
+        window.set_accept_focus(True)
+
+        if self._parent_window_xid is not None:
+            display = Gdk.Display.get_default()
+            parent = GdkX11.X11Window.foreign_new_for_display(
+                display, self._parent_window_xid)
+            window.set_transient_for(parent)
+
+    def _set_decorated(self, decorated):
+        '''
+        To disable default window decorations.
+
+        Args:
+            decorated (bool): False tells window manager not to decorate the window.
+
+        Returns:
+            None
+        '''
+        self._decorated = decorated
+        self.set_decorated(self._decorated)
+
+    def _set_position(self, position):
+        '''
+        Sets a position constraint for this window.
+
+        Args:
+            position (Gtk.WindowPosition): a position constraint
+
+        Returns:
+            None
+        '''
+        self._position = position
+        self.set_position(self._position)
+
+    def _set_border_width(self, border_width):
+        '''
+        Sets border width for window i.e amount of space to leave around
+        the outside of the container
+
+        Args:
+            border_width (int): Width of border
+
+        Returns:
+            None
+        '''
+        self._border_width = border_width
+        self.set_border_width(self._border_width)
+
+    def _set_has_resize_grip(self, has_resize_grip):
+        '''
+        Sets whether window has a corner resize grip.
+
+        Args:
+            has_resize_grip (bool): True if window can be resized
+
+        Returns:
+            None
+        '''
+        self._has_resize_grip = has_resize_grip
+        self.set_has_resize_grip(self._has_resize_grip)
+
+    def _add_events(self, event):
+        '''
+        Adds the events to the event mask for the window.
+
+        Args:
+            event (Gdk.EventMask): an event mask
+        Returns:
+            None
+        '''
+        self._events = event
+        if self._events is not None:
+            self.add_events(self._events)
```

Comment:
Cant we just use add_events and get_events here?

Also, the way that you are maintaining the `_events` variable is broken.  You are overwritting, no appending (this is add events, not set events).  Please just use add_events and get_events.